internal/datastore/postgres/watch.go

```diff
@@ -235,18 +292,122 @@ func (pgd *pgDatastore) loadChanges(ctx context.Context, revisions []revisionWit
 		}
 
 		if _, found := filter[createdXID.Uint64]; found {
-			tracked.AddChange(ctx, txidToRevision[createdXID.Uint64], nextTuple, core.RelationTupleUpdate_TOUCH)
+			if err := tracked.AddRelationshipChange(ctx, txidToRevision[createdXID.Uint64], nextTuple, core.RelationTupleUpdate_TOUCH); err != nil {
+				return err
+			}
 		}
 		if _, found := filter[deletedXID.Uint64]; found {
-			tracked.AddChange(ctx, txidToRevision[deletedXID.Uint64], nextTuple, core.RelationTupleUpdate_DELETE)
+			if err := tracked.AddRelationshipChange(ctx, txidToRevision[deletedXID.Uint64], nextTuple, core.RelationTupleUpdate_DELETE); err != nil {
+				return err
+			}
 		}
 	}
 	if changes.Err() != nil {
-		return nil, fmt.Errorf("unable to load changes for XID: %w", err)
+		return fmt.Errorf("unable to load changes for XID: %w", err)
 	}
+	return nil
+}
 
-	reconciledChanges := tracked.AsRevisionChanges(func(lhs, rhs uint64) bool {
-		return filter[lhs] < filter[rhs]
-	})
-	return reconciledChanges, nil
+func (pgd *pgDatastore) loadNamespaceChanges(ctx context.Context, min uint64, max uint64, txidToRevision map[uint64]revisionWithXid, filter map[uint64]int, tracked *common.Changes[revisionWithXid, uint64]) error {
+	sql, args, err := queryChangedNamespaces.Where(sq.Or{
+		sq.And{
+			sq.LtOrEq{colCreatedXid: max},
+			sq.GtOrEq{colCreatedXid: min},
+		},
+		sq.And{
+			sq.LtOrEq{colDeletedXid: max},
+			sq.GtOrEq{colDeletedXid: min},
+		},
```

Comment:
can we refactor this and turn it into a function? I bet it's used in various places

Reply:
It used twice. I'd rather leave as-is unless you feel strongly we need to extract it